### ISSUE TITLE
Semantic Token Lost with Lombok Annotations in Enum Due to Conflict with Eclipse Java Plugin in VS Code

### ISSUE THREAD

**Issue by Skyxim** [spring-projects/sts4#1383]
**Describe the bug**
When using Lombok annotations with an enum in VS Code, the semantic token is lost, leading to missing syntax highlighting and unresolved identifiers. This issue occurs when the Spring Boot Tools plugin is enabled. The issue does not occur when the plugin is disabled or when Lombok annotations are not used with the enum.

**To Reproduce**

  

1. Install VS Code and enable both the Spring Boot Tools plugin and the Lombok plugin.

2. Create a new Java project and add an enum class with a Lombok annotation.

  
```
import lombok.Getter;

  

public enum Status {

    ACTIVE,

    INACTIVE;

  

    @Getter

    private final String description = "status";

}
```

  

3. Open this project in VS Code. You will notice the semantic token (e.g., syntax highlighting, identifier resolution) is not functioning properly.

4. Disable the Spring Boot Tools plugin and restart VS Code. Semantic tokens should now work as expected.

5. If you remove the Lombok annotations from the enum, the issue does not occur, even with the Spring Boot Tools plugin enabled.

**Comment by martinlippert** [comment]
As a quick workaround, you can disable the syntax highlighting for embedded languages in the preferences `Spring Boot Tools -> Features -> Embedded Syntax Highlighting`. That should bring back the old behavior and you can continue to use all the other Spring Boot Tools features - instead of disabling the extension altogether.

**Comment by BoykoAlex** [comment]
Fixed via b6cb5d0690cc38e9d930c54a113e4b02f304b864
